python/res/job_queue/job_queue_manager.py

```diff
@@ -122,27 +123,29 @@ def __repr__(self):
         return 'JobQueueManager(waiting=%d, running=%d, success=%d, failed=%d, %s)' % (nw,nr,ns,nf,ir)
 
     def max_running(self):
-        return self.queue.get_max_running()
+        return max(1, self.queue.get_max_running())
```

Comment:
Please elaborate.

Reply:
When running a single test run, a local copy of QueueConfig is made, and used to create the JobQueue. The local copy does not include the queue options set, which basically is MAX_RUNNING. Thus max running is set to 0 in the driver. 

The old system has an awkward workaround, when max running was 0:

```
  int max_running = job_queue_get_max_running(queue);
  int num_submit_new = util_int_min(max_submit,  max_running - total_active);

  // If max_running == 0 that should be interpreted as no limit; i.e. the queue
  // layer will attempt to send an unlimited number of jobs to the driver - the
  // driver can reject the jobs.
  if (max_running == 0)
    num_submit_new = util_int_min(max_submit,
                                  job_queue_status_get_count(queue->status,
                                                             JOB_QUEUE_WAITING));
```

I think we can reiterate over this when we define the API for the queue system